### PR TITLE
handle headers as both strings and arrays

### PR DESCRIPTION
**Pull request by kellyselden**
When I don't supply the express response in the visit options, everything works fine, but when I do supply it:

``` js
app.visit(url, {
  response: res
})
```

The headers parsing fails because sometimes the header values are already arrays and don't have the split function. A sample while debugging:

```
// values masked...
debug> exec headers
{ 'set-cookie': 
   [ 'ident=14xxxxxxxxx24; Max-Age=1800; Doma... (length: 160)',
     'session=0xxxxxxxxD; Domain=test.com; Path=/;... (length: 89)' ],
  'x-session-id': '5xxxxxxxxx4',
  'x-registration-id': '8xxxxxxx4',
  'x-appserver-id': 'test',
  token: 'eyJhbGciOiJIUzxxxxxxxxxxxxxxxxxRDVGQkR... (length: 1636)' }
```

I don't yet know if this is an issue only I would have for some reason, or it will start turning up in other apps.

### CODE REVIEW

**Comment by kellyselden** [comment]
I should add that I'm using the standalone fastboot package, without any middleware.

File: src/fastboot-headers.js [hunk 1]
```diff
@@ -5,7 +5,14 @@ function FastBootHeaders(headers) {
   this.headers = {};
 
   for (var header in headers) {
-    this.headers[header] = headers[header].split(', ');
+    let value = headers[header];
+
+    // convert to array if not already
+    if (value.split) {
```
Comment (arjansingh) [inline src/fastboot-headers.js]: I'd prefer doing an `instanceof` or `typeof` check here instead.

Comment (kellyselden) [inline src/fastboot-headers.js]: changed.

**Comment by kellyselden** [comment]
I finally found the evidence needed to get this merged. Express turns the `set-cookie` response header into an array if you set more than one cookie:

https://github.com/expressjs/express/blob/bd47aeb88d28c6cff6e2d8e5378ec73c262e039f/lib/response.js#L805
https://github.com/expressjs/express/blob/bd47aeb88d28c6cff6e2d8e5378ec73c262e039f/lib/response.js#L682

**Comment by kellyselden** [comment]
cc @danmcclain and @tomdale, I remember on the call you two expressed concern over this change.

**Comment by rwjblue** [comment]
closed by https://github.com/ember-fastboot/fastboot/pull/113

**Pull request closed without merging**